source/Planet.h

```diff
@@ -37,6 +37,39 @@ class System;
 // are available, as well as attributes that determine what sort of missions
 // might choose it as a source or destination.
 class Planet {
+public:
+	class Port
+	{
+	public:
+		// The different properties that can be recharged by a port.
+		enum RechargeType
+		{
+			None = 0x0,
+			Shields = 0x1,
+			Hull = 0x2,
+			Energy = 0x4,
+			Fuel = 0x8,
```

Comment:
I'd use bit shifts instead of hex values.
```suggestion
			None = 0,
			Shields = (1 << 0),
			Hull = (1 << 1),
			Energy = (1 << 2),
			Fuel = (1 << 3),
```
(This is what ShipEvent uses.)